File: Csharp.Controller.Api/Routes/TeacherRoutes.cs
using Csharp.Controller.Api.Controllers;

namespace Csharp.Controller.Api.Routes;

/// <summary>
/// Defines the routes to the teacher endpoints.
/// </summary>
internal static class TeacherRoutes
{
    /// <summary>
    /// The route to <see cref="TeachersController.Add"/>.
    /// </summary>
    public const string Add = Base;
    
    /// <summary>
    /// The base route to access the endpoints
    /// of the <see cref="TeachersController"/>.
    /// </summary>
    public const string Base = "/api/Teachers";
    
    /// <summary>
    /// The route to <see cref="TeachersController.Delete"/>. 
    /// </summary>
    public const string Delete = Get;
    
    /// <summary>
    /// The route to <see cref="TeachersController.Get"/>.
    /// </summary>
    public const string Get = $"{Base}/{{id:guid}}";

    /// <summary>
    /// The route to <see cref="TeachersController.GetAll"/>.
    /// </summary>
    public const string GetAll = Base;

    /// <summary>
    /// The route to <see cref="TeachersController.Update"/>.
    /// </summary>
    public const string Update = Get;
}

File: Fsharp.Controller.Api/Routes/TeacherRoutes.fs
module Fsharp.Controller.Api.Routes.Routes

[<RequireQualifiedAccess>]
module TeacherRoutes =

    [<Literal>]
    let Base = "/api/Teachers"
    
    [<Literal>]
    let Add = Base
    
    [<Literal>]
    let Get = Base + "/{id:guid}"
    
    [<Literal>]
    let GetAll = Base
    
    [<Literal>]
    let Delete = Get
    
    [<Literal>]
    let Update = Get
